modules/pre-commit.nix

```diff
@@ -325,7 +340,36 @@ in
                 echo 1>&2 "    3. add .pre-commit-config.yaml to .gitignore"
               else
                 ln -s ${configFile} .pre-commit-config.yaml
-                pre-commit install
+                # Remove any previously installed hooks (since pre-commit itself has no convergent design)
+                hooks="pre-commit pre-merge-commit pre-push prepare-commit-msg commit-msg post-checkout post-commit"
+                for hook in $hooks; do
+                  pre-commit uninstall -t $hook
+                done
+                # Add hooks for configured stages (only) ...
+                if [ ! -z "${concatStringsSep " " install_stages}" ]; then
+                  for stage in ${concatStringsSep " " install_stages}; do
+                    if [[ "$stage" == "manual" ]]; then
+                      continue
+                    fi
+                    case $stage in
+                      # if you amend these switches please also review $hooks above
+                      commit | merge-commit | push)
```

Comment:
Why do we diverge from pre-commit settings by not just specifying pre-push instead of push?

Reply:
Precommit has two identifiers "push" (the stage) and "pre-push" (the hook) in an attempt to be extra "correct" about things, while achieving only plain confusion. Here it's "push" (the stage).

Reply:
Ps: I also wondered and got bitten. More than once.